images/ami_kms_key.tf

```diff
@@ -0,0 +1,81 @@
+# ------------------------------------------------------------------------------
+# Create the KMS key for encrypting AMIs in the Images account
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "ami_kms_doc" {
+  statement {
+    sid = "Enable IAM User Permissions"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.images.account_id}:root"]
+    }
+
+    actions = [
+      "kms:*",
+    ]
+
+    resources = ["*"]
```

Comment:
It might be possible to limit this to only this particular KMS key. I'm unsure about this.